packages/truffle/test/sources/mixed_testing/test/js-test.js

```diff
@@ -0,0 +1,8 @@
+const StrangeEventTest = artifacts.require("StrangeEventTest");
+
+contract("StrangeEventTest", function () {
+  it("should emit a strange event", async function () {
+    const instance = await StrangeEventTest.deployed();
+    await instance.run();
```

Comment:
should this have an assertion or is this just here to test that it doesn't throw or something?

Reply:
Ah, sorry, that perhaps warrants some explanation.  That's not a test of our code at all.  That's a test *within the Truffle project which is a test fixture*.  The test of our code is elsewhere; it runs `truffle test --show-events`, causing this test to run, and then and checks that the correct things get logged as a result.